cmake/modules/BuildOpentelemetry.cmake

```diff
@@ -0,0 +1,87 @@
+function(target_create _target _lib)
+  add_library(${_target} STATIC IMPORTED)
+  set_target_properties(
+    ${_target} PROPERTIES IMPORTED_LOCATION
+                          "${opentelemetry_BINARY_DIR}/${_lib}")
+endfunction()
+
+function(build_opentelemetry)
+  set(opentelemetry_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/opentelemetry-cpp")
+  set(opentelemetry_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/opentelemetry-cpp")
+  set(opentelemetry_cpp_targets opentelemetry_trace opentelemetry_exporter_jaeger_trace)
+  set(opentelemetry_CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+                               -DWITH_JAEGER=ON
+                               -DBUILD_TESTING=OFF
+                               -DWITH_EXAMPLES=OFF
+                               -DBoost_INCLUDE_DIR=${CMAKE_BINARY_DIR}/boost/include)
+
+  set(opentelemetry_libs
+      ${opentelemetry_BINARY_DIR}/sdk/src/trace/libopentelemetry_trace.a
+      ${opentelemetry_BINARY_DIR}/sdk/src/resource/libopentelemetry_resources.a
+      ${opentelemetry_BINARY_DIR}/sdk/src/common/libopentelemetry_common.a
+      ${opentelemetry_BINARY_DIR}/exporters/jaeger/libopentelemetry_exporter_jaeger_trace.a
+      ${opentelemetry_BINARY_DIR}/ext/src/http/client/curl/libhttp_client_curl.a
+      ${CURL_LIBRARIES}
+  )
+  set(opentelemetry_include_dir ${opentelemetry_SOURCE_DIR}/api/include/
+                                ${opentelemetry_SOURCE_DIR}/exporters/jaeger/include/
+                                ${opentelemetry_SOURCE_DIR}/ext/include/
+                                ${opentelemetry_SOURCE_DIR}/sdk/include/)
+  include_directories(SYSTEM ${opentelemetry_include_dir})
+  # TODO: add target based propogation
+  set(opentelemetry_deps opentelemetry_trace opentelemetry_resources opentelemetry_common
+                         opentelemetry_exporter_jaeger_trace http_client_curl
+			 ${CURL_LIBRARIES})
+
+  if(CMAKE_MAKE_PROGRAM MATCHES "make")
+    # try to inherit command line arguments passed by parent "make" job
+    set(make_cmd $(MAKE) ${opentelemetry_cpp_targets})
+  else()
+    set(make_cmd ${CMAKE_COMMAND} --build <BINARY_DIR> --target
+                 ${opentelemetry_cpp_targets})
+  endif()
+
+  if(NOT WITH_SYSTEM_BOOST)
+    list(APPEND dependencies Boost)
+  endif()
+
+  include(ExternalProject)
+  ExternalProject_Add(
+    opentelemetry-cpp
+    GIT_REPOSITORY https://github.com/ideepika/opentelemetry-cpp.git
```

Comment:
we might want to use ceph repo here, @yuvalif do you have previlage to fork opentelemetry-cpp to ceph org?

Reply:
why do we need a fork?
why not the official repo?

Reply:
need this extra patch https://github.com/ideepika/opentelemetry-cpp/commit/d02acef1fe31e6bc1f60e378bdb0a94ea7d80a26 since thrift-devel installs system boost and when we use that version of boost with ceph compiled boost, it leads to failure due to 2 different versions of boost in use.
Not sure if it will be opentelemetry specific issue, but I can send a patch to opentelemetry, later this week; for the mean time, we can either patch this or use branch wip-ceph

Reply:
lets try to get that patch upstream.
if they don't accept it, or it takes too much time, we will create our own fork
in the meantime we can use your fork (so, we don't delay this PR)

Reply:
yes!